### PR TITLE
refactor async to use apollo

### PR DESCRIPTION
**Pull request by jbaxleyiii**

### CODE REVIEW

File: examples/async/src/index.js [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
-if (process.env.NODE_ENV !== 'production') {
-  middleware.push(createLogger());
-}
+import { ApolloClient } from 'apollo-client';
+import { withClientState } from 'apollo-link-state';
+import { InMemoryCache } from 'apollo-cache-inmemory';
+import { ApolloProvider } from 'react-apollo';
+
+import App from './containers/App';
+import schema from './schema';
```
Comment (peggyrayzis) [inline examples/async/src/index.js:10]: Would it be more accurate to call this a resolver map?

File: examples/async/src/schema/index.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  Query: {
+    // this is only called on a cache miss
+    postsBySubreddit: async (_, { subreddit }) => {
+      try {
+        // trigger request posts
+        const items = await fetch(`https://www.reddit.com/r/${subreddit}.json`)
+          .then(response => response.json())
+          .then(json => json.data.children.map(child => child.data))
+          .then(posts =>
+            posts.map(({ title }) => ({ title, __typename: 'Post' })),
```
Comment (peggyrayzis) [inline examples/async/src/schema/index.js:32]: What's our plan for typechecking? Do these types correspond to types on the server schema?

**Review by peggyrayzis — APPROVED** [review]
Two minor questions but overall, this is SO awesome!! 😍

File: examples/async/src/index.js (hunk 1 above)
Comment (jbaxleyiii) [reply examples/async/src/index.js:10]: probably!

File: examples/async/src/schema/index.js (hunk 2 above)
Comment (jbaxleyiii) [reply examples/async/src/schema/index.js:32]: none right now haha. Which is on par with existing state solutions but I would like to have it

Comment (peggyrayzis) [reply examples/async/src/schema/index.js:32]: let's figure out a plan in our 1:1, i think people are going to ask about it when we release

**Pull request closed without merging**